src/main/java/org/broadinstitute/dsm/route/DownloadPDFRoute.java

```diff
@@ -70,434 +37,43 @@ public Object processRequest(Request request, Response response, String userId)
                 if (request.url().contains(RoutePath.DOWNLOAD_PDF)) {
-                    String requestBody = request.body();
                     if (StringUtils.isNotBlank(requestBody)) {
-                        JSONObject jsonObject = new JSONObject(requestBody);
-                        String userIdR = (String) jsonObject.get(RequestParameter.USER_ID);
-                        Integer userIdRequest = Integer.parseInt(userIdR);
-                        if (!userId.equals(userIdR)) {
-                            throw new RuntimeException("User id was not equal. User Id in token " + userId + " user Id in request " + userIdR);
-                        }
-
-                        String ddpParticipantId = (String) jsonObject.get(RequestParameter.DDP_PARTICIPANT_ID);
-                        if (StringUtils.isNotBlank(ddpParticipantId)) {
-                            String configName = null;
-                            if (jsonObject.has(RequestParameter.CONFIG_NAME)) {
-                                configName = (String) jsonObject.get(RequestParameter.CONFIG_NAME);
-                            }
-                            String medicalRecord = null;
-                            if (jsonObject.has("medicalRecordId")) {
-                                medicalRecord = (String) jsonObject.get("medicalRecordId");
-                            }
-                            List<String> oncHistoryIDs = null;
-                            if (jsonObject.has("requestId")) {
-                                oncHistoryIDs = new ArrayList<>();
-                                JSONArray array = (JSONArray) jsonObject.get("requestId");
-                                for(int i = 0; i < array.length(); i++){
-                                    oncHistoryIDs.add((String) array.get(i));
-                                }
-                            }
-                            List<PDF> pdfs = null;
-                            if (jsonObject.has("pdfs")) {
-                                pdfs = Arrays.asList(new Gson().fromJson((String) jsonObject.get("pdfs"), PDF[].class));
-                            }
-                            getPDFs(response, ddpParticipantId, configName, medicalRecord, oncHistoryIDs, pdfs, realm, requestBody, userIdRequest);
-                            return new Result(200);
-                        }
-                        else {
-                            response.status(500);
-                            throw new RuntimeException("Error missing ddpParticipantId");
-                        }
+                        Long userIdRequest = Long.parseLong((String) new JSONObject(requestBody).get(RequestParameter.USER_ID));
+                        DownloadPDF downloadPDFRequest = new DownloadPDF(requestBody);
+                        UserDto user = new UserDao().get(userIdRequest).orElseThrow();
+                        return downloadPDFRequest.getPDFs(response, user, realm, requestBody);
```

Comment:
So the basic idea here is to separate downloading from participant page and getting the list of available pdfs for miscellaneous download. 
In order to do the first, I create a member of the class `DownloadPDF` and pass the request body to that class to decide what kind of pdf(s) to return